src/parser/jobs/mch/modules/Reassemble.tsx

```diff
@@ -1,86 +1,108 @@
+import {Plural, Trans} from '@lingui/react'
+import {DataLink} from 'components/ui/DbLink'
+import {ActionKey} from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
 import {Data} from 'parser/core/modules/Data'
-import Suggestions, {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
 import React from 'react'
 
 // These are the only GCDs that should be reassembled under normal circumstances
-const REASSEMBLE_GCDS: Array<keyof ActionRoot> = [
+const REASSEMBLE_GCDS: ActionKey[] = [
 	'AIR_ANCHOR',
 	'AUTO_CROSSBOW',
 	'DRILL',
 	'SPREAD_SHOT',
+	// 6.0:
+	//'CHAIN_SAW',
+	//'SCATTER_GUN',
 ]
 
-const OTHER_GCDS: Array<keyof ActionRoot> = [
+const OTHER_GCDS: ActionKey[] = [
 	'BIOBLASTER',
 	'HEAT_BLAST',
 	'HEATED_SPLIT_SHOT',
 	'HEATED_SLUG_SHOT',
 	'HEATED_CLEAN_SHOT',
 ]
 
-export default class Reassemble extends Module {
+export class Reassemble extends Analyser {
 	static override handle = 'reassemble'
 
-	@dependency private combatants!: Combatants
 	@dependency private data!: Data
 	@dependency private suggestions!: Suggestions
 
 	private badReassembles = 0
 	private droppedReassembles = 0
-	private lastUse = 0
+	private lastGcd = 0
+	private reassembleActive = false
 
 	private REASSEMBLE_GCDS: number[] = []
 	private OTHER_GCDS: number[] = []
 
-	protected override init() {
+	override initialise() {
 		this.REASSEMBLE_GCDS = REASSEMBLE_GCDS.map(key => this.data.actions[key].id)
 		this.OTHER_GCDS = OTHER_GCDS.map(key => this.data.actions[key].id)
 
-		this.addEventHook('cast', {by: 'player', abilityId: [...this.REASSEMBLE_GCDS, ...this.OTHER_GCDS]}, this.onCast)
-		this.addEventHook('removedebuff', {by: 'player', abilityId: STATUSES.REASSEMBLED.id}, this.onRemoveReassemble)
+		const reassembleFilter = filter<Event>()
+			.source(this.parser.actor.id)
+			.status(this.data.statuses.REASSEMBLED.id)
+
+		const gcdFilter = filter<Event>()
+			.source(this.parser.actor.id)
+			.type('action')
+			.action(oneOf([...this.REASSEMBLE_GCDS, ...this.OTHER_GCDS]))
+
+		this.addEventHook(reassembleFilter.type('statusApply'), this.onReassemble)
+		this.addEventHook(gcdFilter, this.onCast)
+		this.addEventHook(reassembleFilter.type('statusRemove'), this.onRemove)
 		this.addEventHook('complete', this.onComplete)
 	}
 
-	private onCast(event: CastEvent) {
-		if (this.combatants.selected.hasStatus(STATUSES.REASSEMBLED.id)) {
-			this.badReassembles += this.REASSEMBLE_GCDS.includes(event.ability.guid) ? 0 : 1
+	private onReassemble() {
+		if (this.reassembleActive) {
+			this.droppedReassembles++
+		}
+		this.reassembleActive = true
+	}
+
+	private onCast(event: Events['action']) {
+		if (this.reassembleActive && !this.REASSEMBLE_GCDS.includes(event.action)) {
+			this.badReassembles += 1
```

Comment:
Any reason not to `this.badReassembles++`? I guess future proofing if you get stacks later or something, but for now that's probably clearer unless something breaks.

Reply:
`+= 1` supremacy, there's no world where `++`  is clearer than `+= 1`.

(this is what happens when you spend too much time with python)